cpu/sam0_common/periph/uart.c

```diff
@@ -69,18 +69,18 @@ int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, void *arg)
 
     /* reset the UART device */
     dev(uart)->CTRLA.reg = SERCOM_USART_CTRLA_SWRST;
-    while (dev(uart)->SYNCBUSY.reg & SERCOM_USART_SYNCBUSY_SWRST) {}
+    while (dev(uart)->SYNCBUSY.bit.SWRST) {}
```

Comment:
Actually, I like the original check much better, but this might be subjective. At some point we decided to go the the `.reg & ..` notation consistently for all the sam0 periph drivers...

Reply:
with the updated CMSIS headers from the newer ASF the bit fields are read-only, and `reg` is RW. So I would guess the compiler could optimise things a bit (more) when using read-only registers for read operations on the bit-field instead of using an RW register when using `reg`. 

Do I make sense here?

Reply:
Plus, I would therefore recommend to adapt all periph drivers accordingly to have it consistent again.

Reply:
Yepp, code size does not change with this, re-checked it. So yes, with them being read only why not change it. Maybe this would be cleaner done in a separate PR, but just leave it here if you want.